### ISSUE TITLE
ArgumentNullException after destroying entities

### ISSUE THREAD

**Issue by Jaden-Giordano** [craftworkgames/MonoGame.Extended#498]
### Error:

Message | Type
-- | -- |
{"Value cannot be null.\r\nParameter name: key"} | System.ArgumentNullException

This occurs after running `Destroy()` on an entity.

### Stack Trace:
```   at System.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at MonoGame.Extended.Entities.EntityManager.ProcessMarkedEntitiesWith(List`1 processingSystems) in C:\BuildAgent\work\b0358f41d1a3fba\Source\MonoGame.Extended.Entities\EntityManager.cs:line 256
   at MonoGame.Extended.Entities.EntityComponentSystem.Update(GameTime gameTime) in C:\BuildAgent\work\b0358f41d1a3fba\Source\MonoGame.Extended.Entities\EntityComponentSystem.cs:line 259
   at WisePalm.MainGame.Update(GameTime gameTime) in D:\Projects\Games\WisePalm\WisePalm\MainGame.cs:line 86
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime)
   at Microsoft.Xna.Framework.Game.Tick()
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop()
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior)
   at WisePalm.Program.Main() in D:\Projects\Games\WisePalm\WisePalm\Program.cs:line 17```

**Comment by LithiumToast** [comment]
Can you provide a snippet of your own code?

**Comment by Jaden-Giordano** [comment]
<em>MainGame.cs : Update()</em>
```cs
int count = 0;
protected override void Update(GameTime gameTime) {
    count += 1;
    if (count == 3) {
        sceneService.Load("Second"); // Entity destruction happens in here.
    }

    base.Update(gameTime);

    sceneService.Update(gameTime);

    ecs.Update(gameTime); // Error thrown here.
}
```

<em>SceneService.cs : Load()</em>
```cs
public void Load(string sceneName) {
    Current?.Destroy(); // Entity destruction happens in here.

    Current = sceneFactory.CreateScene(sceneName);
}
```

<em>Scene.cs : Destory()</em>
```cs
public virtual void Destroy() {
    foreach (var e in activeEntities)
        e.Destroy(); // Destory entity here.
}
```

**Comment by LithiumToast** [comment]
I don't quite understand the stack trace. 

The jump from `ProcessMarkedEntitiesWith` to `TryGetValue` is not obvious. Furthermore, [line 256](https://github.com/craftworkgames/MonoGame.Extended/blob/develop/Source/MonoGame.Extended.Entities/EntityManager.cs#L256) is not doing anything with a dictionary...

What version of the repository are you using?

**Comment by Jaden-Giordano** [comment]
I'm away for the weekend I'll let you know when I'm back, thanks!

**Comment by rafaelalmeidatk** [comment]
This problem is already solved by #489, are you using the latest version of develop branch?

**Comment by Navomi-Jaden** [comment]
Im using the latest release so I guess not.

**Comment by craftworkgames** [comment]
This would have been included in the [version 1.1](https://www.nuget.org/packages?q=monogame.extended) release I did a few days ago.
